prowler/lib/check/custom_checks_metadata.py

```diff
@@ -0,0 +1,53 @@
+import sys
+
+import yaml
+from schema import Schema
+
+from prowler.lib.logger import logger
+
+valid_severities = ["critical", "high", "medium", "low", "informational"]
+custom_checks_metadata_schema = Schema(
+    {
+        "Checks": {
+            str: {
+                "Severity": str,
+            }
+        }
+    }
+)
+
+
+def parse_custom_checks_metadata_file(provider: str, parse_custom_checks_metadata_file):
+    try:
+        with open(parse_custom_checks_metadata_file) as f:
+            custom_checks_metadata = yaml.safe_load(f)["CustomChecksMetadata"][provider]
+            custom_checks_metadata_schema.validate(custom_checks_metadata)
+        return custom_checks_metadata
+    except Exception as error:
+        logger.critical(
+            f"{error.__class__.__name__} -- {error}[{error.__traceback__.tb_lineno}]"
+        )
+        sys.exit(1)
+
+
+def update_checks_metadata(bulk_checks_metadata, custom_checks_metadata):
+    try:
+        # Update checks metadata from CustomChecksMetadata file
+        for check, custom_metadata in custom_checks_metadata["Checks"].items():
+            check_metadata = bulk_checks_metadata.get(check)
+            if check:
+                bulk_checks_metadata[check] = update_check_metadata(
+                    check_metadata, custom_metadata
+                )
```

Comment:
Can we check if check_metadata is present here for given check .
User can give some `check = "some_random_check_name"` which aren't existed then
`check_metadata = bulk_checks_metadata.get(check)` becomes `None` as check is not existed.
It will then try updating a check which isn't existed in next line.

What's your thoughts on this?

Reply:
I understand your point but we are looping over the `custom_checks_metadata` so we only modify the checks present in that file regardless of user input. If you set `--checks non_existent_check` nothing happens there since that function will only update the checks with the modified metadata.

Reply:
```bash
prowler aws --custom-checks-metadata-file prowler/config/custom_checks_metadata.yaml -s s3 --log-level DEBUG
```
I just tried in my local as well 

 It gave below error. 
`[File: custom_checks_metadata.py:48]    [Module: custom_checks_metadata]         CRITICAL: AttributeError -- 'NoneType' object has no attribute 'Severity'[42]`

When I updated the condition from `if check:` to `if check_metadata:` it worked.

Reply:
What is the content of the `custom_checks_metadata.yaml`?

Reply:
```yaml
CustomChecksMetadata:
  aws:
    Checks:
      "some_randome_check":
        Severity: "high"
      
```

Reply:
Now I get it, yes this case it's not handled, also I think test would arise that, I'll fix that tomorrow. Any other thought about the refactor, for use having it to all the providers is simple enough, what do you think?

Reply:
Yes @jfagoagas I also found this approach more simpler.